src/Server/Packets/Handle/HandlePacketChatMessage.cs

```diff
@@ -38,10 +39,20 @@ public override void Handle(Event netEvent, ref PacketReader packetReader)
             // Remove any trailing spaces at start and end
             message = message.Trim();
 
-            // Swear filter
-            foreach (var blockedString in BlockedStrings)
-                message = message.Replace(blockedString, "meowww");
+            private  string[] blockedStringReplacement = { "meowww", "meow", "meeow", "purrr", "meowwww", "pur" };
+            
+            if(useProfanityFilter){
+            foreach (var blockedString in blockedStrings){
 
+                if(preventAllCaps){ message = message.ToLower();}
```

Comment:
If `preventAllCaps` is set to `true` then all messages will be set to lowercase regardless if they send messages like

"Hello World" will become "hello world" <-- (undesired behavior)
"HELLO WORLD" will become "hello world"

Reply:
This should be solved now!